File: src/main/java/com/mateusz/spring_boot_todos/controllers/JwtAuthenticationController.java
package com.mateusz.spring_boot_todos.controllers;

import org.springframework.http.ResponseEntity;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import com.mateusz.spring_boot_todos.DTO.JwtTokenRequest;
import com.mateusz.spring_boot_todos.DTO.JwtTokenResponse;
import com.mateusz.spring_boot_todos.services.JwtTokenService;

@RestController
public class JwtAuthenticationController {

    private final JwtTokenService tokenService;

    private final AuthenticationManager authenticationManager;

    public JwtAuthenticationController(JwtTokenService tokenService,
            AuthenticationManager authenticationManager) {
        this.tokenService = tokenService;
        this.authenticationManager = authenticationManager;
    }

    @PostMapping("/authenticate")
    public ResponseEntity<JwtTokenResponse> createToken(
            @RequestBody JwtTokenRequest jwtTokenRequest) {

        var authenticationToken = new UsernamePasswordAuthenticationToken(
                jwtTokenRequest.username(),
                jwtTokenRequest.password());

        var authentication = authenticationManager.authenticate(authenticationToken);

        var token = tokenService.createToken(authentication);

        return ResponseEntity.ok(new JwtTokenResponse(token));
    }
}

File: src/main/kotlin/com/mateusz/todos/kotlin_sboot_todos_backend/controllers/JwtAuthenticationController.kt
package com.mateusz.todos.kotlin_sboot_todos_backend.controllers

import com.mateusz.todos.kotlin_sboot_todos_backend.dto.JwtTokenRequest
import com.mateusz.todos.kotlin_sboot_todos_backend.dto.JwtTokenResponse
import com.mateusz.todos.kotlin_sboot_todos_backend.services.JwtTokenService
import org.springframework.http.ResponseEntity
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/v1")
class JwtAuthenticationController(
    private val tokenService: JwtTokenService,
    private val authenticationManager: AuthenticationManager
) {

    @PostMapping(
        value = ["/login"],
        consumes = ["application/json", "application/xml"],
        produces = ["application/json", "application/xml"]
    )
    fun createToken(
        @RequestBody jwtTokenRequest: JwtTokenRequest
    ): ResponseEntity<JwtTokenResponse> {
        val authenticationToken = UsernamePasswordAuthenticationToken(
            jwtTokenRequest.username,
            jwtTokenRequest.password
        )

        val authentication = authenticationManager.authenticate(authenticationToken)

        val token: String = tokenService.createToken(authentication)

        return ResponseEntity.ok(JwtTokenResponse(token))
    }
}